### ISSUE TITLE
VSCode Powershell Terminal Hang on Connect-AZAccount

### ISSUE THREAD

**Issue by steevaavoo** [steevaavoo/azure-terraform#4]
Having to connect to AZ from cmder rather than the integrated Terminal - which isn't the end of the world, but I'd love to know what's causing the hang and fix it.

No really inspiring results on my Googling so far... Any thoughts?

**Comment by adamrushuk** [comment]
When I try from VSCode, it outputs a Warning message and won't let me cancel:
![image](https://user-images.githubusercontent.com/11696118/52231307-7cd27300-28b1-11e9-81d2-4b281055b045.png)

However, when I follow that link and enter the unique code, the device login completes without issue.

Must be an issue with your end.

**Comment by adamrushuk** [comment]
Oh hold up, I was using v1.0.1 of the Az PowerShell module before.
Just uninstalled that: 
`Get-InstalledModule -Name Az -AllVersions | Uninstall-Module -Verbose -Force`

then installed latest (v1.2.1):
`Install-Module -Name Az -AllowClobber -Repository PSGallery -Verbose`

Now `Connect-AzAccount` opens a different pop-up window than before.

@steevaavoo see if the latest Az PowerShell module fixes it for you.

**Comment by steevaavoo** [comment]
Just checked to confirm that our versions are aligned:

```> Get-InstalledModule -Name Az -AllVersions

Version    Name                                Repository           Description
-------    ----                                ----------           -----------
1.2.1      Az                                  PSGallery            Microsoft Azure PowerShell - Cmdlets to mana...
```

So then I started vscode in Safe Mode, and still got a hang on `Connect-AzAccount`

Then I turned off my perimeter firewall and still got a hang (though can't figure why it would be a factor I thought anything was worth a try!)

**Comment by adamrushuk** [comment]
oh wait, I'm using the latest PowerShell Preview VSCode extension (ms-vscode.powershell-preview
Preview).

Try disabling your current PowerShell extension **first**, then search for and install `ms-vscode.powershell-preview`.

**Comment by steevaavoo** [comment]
Well, this fixed it:

I ran: 
`Get-InstalledModule -Name Az -AllVersions | Uninstall-Module -Verbose -Force` 
from cmder

I then opened vscode in Admin mode (from cmder), and ran:
`Install-Module -Name Az -AllowClobber -Repository PSGallery -Verbose` 
from the vscode terminal.

After that, Connect-AzAccount works in either console, in either Admin or Standard mode, spawning the required login window.

Repeated the process on the laptop but ran the uninstall _and_ reinstall in an elevated VSCode, and that _didn't_ work. Will try the exact same process as above (uninstalling inside cmder then spawning vscode from there and installing)...

~Same hang.~

~So the PC now works, but the laptop vscode terminal still hangs on Connect-AzAccount.~

It _does_ work. The Azure Connect window is spawning behind the VSCode window on the laptop. But it spawns on top on the PC.

Bizarre.

**Comment by steevaavoo** [comment]
After a reboot of both devices, I have confirmed that it works in both cases - however; the Azure MFA connection window appears _behind_ the vscode window which, if you're running fullscreen, or if the vscode window happens to cover the part of the screen where the MFA window spawns, you might believe it's hung.

That may have been the case all along - I don't accurately remember whether or not I went hunting for the MFA window before today.

My final point then, is that this _is_ a bug, because a window that needs input from a user should appear on top - especially if it manifests as a console "hang" when the requisite input window spawns in the background.
